### ISSUE TITLE
Backslashes confuse CMake's dependency scanning

### ISSUE THREAD

**Issue by maharmstone** [mstorsjo/msvc-wine#29]
CMake's dependency scanning for MSVC involves it passing the `/showIncludes` flag to `cl`, which results in it printing 

```
Note: including file: \opt\msvc\vc\tools\msvc\14.29.30133\include\string
Note: including file:  \opt\msvc\vc\tools\msvc\14.29.30133\include\yvals_core.h
Note: including file:   \opt\msvc\vc\tools\msvc\14.29.30133\include\vcruntime.h
```

etc. to stdout. Unfortunately the backslashes confuse it, meaning that it ends up recompiling the whole project every time.

Changing the third line of `/opt/msvc/bin/x64/cl` to

```
$(dirname $0)/wine-msvc.sh $BINDIR/cl.exe "$@" | sed "s/\\\\/\//g" -
```

fixes this, but it's not a very elegant way of doing it. It probably should say "for every line beginning 'Note: including file:', replace backslashes with slashes" - however you express that in sed.

**Comment by mstorsjo** [comment]
Have a look at (the topmost commit in) https://github.com/mstorsjo/msvc-wine/commits/cmake - that contains an example of building a lightly patched cmake and ninja, which then should work fine with respect to dependency tracking. Unfortunately, those patches aren't entirely fit to upstream...

**Comment by mstorsjo** [comment]
Thanks! I actually have a similar patch in a local test branch since some time (with a few minor tweaks to the regex, mine is `'/^Note:/s,\\,/,g'` (but I had applied it in `wine-msvc.sh`, while keeping it in the `cl` wrapper only is a better idea.

I'll try to refresh my memory with what I concluded about it and why I didn't go forward with it, whether it interfered with some of my other build setups or not... But meanwhile, I don't mind taking a patch for it!

This issue was discussed a bit in https://github.com/ninja-build/ninja/issues/1735, and in https://github.com/ninja-build/ninja/issues/1735#issuecomment-604089405 I seem to have concluded that it seemed to work fine that far...

**Comment by darksylinc** [comment]
Hi! First: FANTASTIC WORK! It's amazing.

Second: Happy new year.

Third: I'm hit by this issue and the proposed solution works (i.e. ninja no longer starts from scratch) but I'm seeing a similar problem with RC. The generated rules.ninja contain:

```ninja
#############################################
# Rule for compiling RC files.

rule RC_COMPILER__OgreMain_Release
  depfile = $DEP_FILE
  deps = gcc
  command = "" RC $in $DEP_FILE $out "Note: including file: " "/opt/msvc/bin/x64/cl" /opt/msvc/bin/x64/rc $DEFINES $INCLUDES $FLAGS /fo $out $in
  description = Building RC object $out
```

Which works if I manually edit it to:

```ninja
#############################################
# Rule for compiling RC files.

rule RC_COMPILER__OgreMain_Release
  depfile = $DEP_FILE
  deps = gcc
  command = /opt/msvc/bin/x64/rc $DEFINES $INCLUDES $FLAGS /fo $out $in
  description = Building RC object $out
```

But obviously it reverts if I run CMake again. Any ideas? I tried the suggested solution by patching rc but it doesn't seem to work. Maybe the syntax for sed must be different

**Comment by mstorsjo** [comment]
The original issue here, about ninja not handling the dependencies, should be fixed by baf2d3c637f2b84ada70634a501617d12d1a0b6c.
